src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -58,6 +62,54 @@
 
     private SigningKeyResolver signingKeyResolver;
 
+    Claims expectedClaims = new DefaultClaims();
+
+    @Override
+    public JwtParser expectIssuedAt(Date issuedAt) {
+        if (issuedAt != null) {
+            expectedClaims.setIssuedAt(issuedAt);
+        }
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectIssuer(String issuer) {
+        expect(Claims.ISSUER, issuer);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectAudience(String audience) {
+        expect(Claims.AUDIENCE, audience);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectSubject(String subject) {
+        expect(Claims.SUBJECT, subject);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expectId(String id) {
+        expect(Claims.ID, id);
+
+        return this;
+    }
+
+    @Override
+    public JwtParser expect(String claimName, Object value) {
+        if (claimName != null && claimName.length() > 0 && value != null) {
```

Comment:
This should throw an exception if claimName is null or length of zero and should throw an exception if the value is null - either scenario means the caller is doing something wrong (invalid API usage).